src/screens/OtherProfile/elements/BioAndChat.js

```diff
@@ -0,0 +1,190 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {Pressable, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
+
+import TextAreaChat from '../../../components/TextAreaChat';
+import ToggleSwitch from '../../../components/ToggleSwitch';
+import useSaveAnonChatHook from '../../../database/hooks/useSaveAnonChatHook';
+import {ANON_PM, SIGNED} from '../../../hooks/core/constant';
+import {colors} from '../../../utils/colors';
+import {fonts} from '../../../utils/fonts';
+import {sendAnonymousDMOtherProfile, sendSignedDMOtherProfile} from '../../../service/chat';
+import {trimString} from '../../../utils/string/TrimString';
+
+const CHANNEL_BLOCKED = 'Channel is blocked';
+
+const BioAndChat = (props) => {
+  const {
+    isAnonimity,
+    bio,
+    openBio,
+    dataMain,
+    isSignedMessageEnabled,
+    showSignedMessageDisableToast,
+    loadingGenerateAnon,
+    avatarUrl,
+    anonProfile,
+    username,
+    toggleSwitch,
+    isAnonimityEnabled
+  } = props;
+  const {saveChatFromOtherProfile, savePendingChatFromOtherProfile} = useSaveAnonChatHook();
+  const [dmChat, setDmChat] = React.useState('');
+  const [loadingSendDM, setLoadingSendDM] = React.useState(false);
+
+  React.useEffect(() => {
+    setDmChat('');
+  }, [isSignedMessageEnabled]);
+
+  const sendSignedDM = async () => {
+    try {
+      setLoadingSendDM(true);
+      const signedMParams = {
+        user_id: dataMain.user_id,
+        message: dmChat
+      };
+      const response = await sendSignedDMOtherProfile(signedMParams);
+      const newResponse = {...response, members: response?.message?.members};
+      await saveChatFromOtherProfile(newResponse, 'sent', true, SIGNED);
+      setDmChat('');
+    } catch (error) {
+      if (__DEV__) {
+        console.log(error, 'error');
+      }
+    } finally {
+      setLoadingSendDM(false);
+    }
+  };
+
+  const sentAnonDM = async () => {
+    try {
+      setLoadingSendDM(true);
+      const {
+        anon_user_info_emoji_name,
+        anon_user_info_emoji_code,
+        anon_user_info_color_name,
+        anon_user_info_color_code
+      } = anonProfile ?? {};
+
+      const anonDMParams = {
+        user_id: dataMain.user_id,
+        message: dmChat,
+        anon_user_info_emoji_name,
+        anon_user_info_emoji_code,
+        anon_user_info_color_name,
+        anon_user_info_color_code
+      };
+      const response = await sendAnonymousDMOtherProfile(anonDMParams);
+      await saveChatFromOtherProfile(response, 'sent', true, ANON_PM);
+      setDmChat('');
+    } catch (e) {
+      if (e?.response?.data?.status === CHANNEL_BLOCKED) {
+        const response = e?.response?.data?.data;
+        await savePendingChatFromOtherProfile(response, true);
+        setDmChat('');
+        return;
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This error handling block is specific to a certain error status (`CHANNEL_BLOCKED`). It would be better to have a more general error handling mechanism that can handle different types of errors.



<!-- This is an auto-generated comment by OSS CodeRabbit -->